astroquery/xmm_newton/core.py

```diff
@@ -0,0 +1,343 @@
+            values: OM, R1, R2, M1, M2, PN
+        instmode : string
+            instrument mode, optional
+            examples: Fast, FlatFieldLow, Image, PrimeFullWindow
+        filter : string
+            filter, optional
+            examples: Closed, Open, Thick, UVM2, UVW1, UVW2, V
+        expflag : string
+            exposure flag, optional, by default everything
+            values: S, U, X(not applicable)
+        expno : integer
+            exposure number with 3 digits, by default all exposures
+            examples: 001, 003
+        name : string
+            product type, optional, 6 characters, by default all product types
+            examples: 3COLIM, ATTTSR, EVENLI, SBSPEC, EXPMAP, SRCARF
+        datasubsetno : character
+            data subset number, optional, by default all
+            examples: 0, 1
+        sourceno : hex value
+            source number, optional, by default all sources
+            example: 00A, 021, 001
+        extension : string
+            file format, optional, by default all formats
+            values: ASC, ASZ, FTZ, HTM, IND, PDF, PNG
+        filename : string
+            file name to be used to store the file, optional, default
+            None
+        verbose : bool
+            optional, default 'False'
+            flag to display information about the process
+
+        Returns
+        -------
+        None. It downloads the observation indicated
+        """
+
+        link = self.data_aio_url + "obsno=" + observation_id
+
+        if level is not None:
+            link = link + "&level=" + level
+
+        if instname is not None:
+            link = link + "&instname=" + instname
+
+        if instmode is not None:
+            link = link + "&instmode=" + instmode
+
+        if filter is not None:
+            link = link + "&filter=" + filter
+
+        if expflag is not None:
+            link = link + "&expflag=" + expflag
+
+        if expno is not None:
+            link = link + "&expno=" + expno
+
+        if name is not None:
+            link = link + "&name=" + name
+
+        if datasubsetno is not None:
+            link = link + "&datasubsetno=" + datasubsetno
+
+        if sourceno is not None:
+            link = link + "&sourceno=" + sourceno
+
+        if extension is not None:
+            link = link + "&extension=" + extension
+
+        response = self._handler.request('GET', link)
+        if response is not None:
+            response.raise_for_status()
+
+            if filename is None:
+                if "Content-Disposition" in response.headers.keys():
+                    filename = re.findall('filename="(.+)"',
+                                          response.headers[
+                                              "Content-Disposition"])[0]
+                else:
+                    filename = observation_id + ".tar"
+
+            self._handler.get_file(filename, response=response,
```

Comment:
see above - you should not define an XMM-specific handler here, you should use astroquery's built-in data retrieval handlers.

Reply:
Updated to retrieve_file